src/components/main/vis/results/ScatterPlot.js

```diff
@@ -1,83 +1,88 @@
-  app: {
-    run: { runID },
-    toggle: {
-      vis: {results: {activeResult, selectedFeature, selectedGroup, availablePlots}}
-    }
-  },
-  actions: {
-    thunks: {
-      fetchScatter
-    }
-  }
-}) => {
-  // use local state for data since too big for redux store
-  const [scatterData, setScatterData] = useState( [] );
-  
-  useEffect(() => {
-    fetchScatter().then((data) => {
-      setScatterData(data);
-    })
-  }, [activeResult])
+    ({
+        app: {
+            run: { runID },
+            toggle: {
+                vis: {
+                    results: { activeResult, selectedFeature, selectedGroup, availablePlots }
+                }
+            }
+        },
+        actions: {
+            thunks: { fetchScatter }
+        }
+    }) => {
+        // use local state for data since too big for redux store
+        const [scatterData, setScatterData] = useState([]);
 
-  useEffect(() => {
-    if(selectedGroup){
-    setScatterData( [] ) // set to loading
-    fetchScatter().then((data) => {
-      setScatterData(data);
-    });
-    }
-  }, [selectedGroup])
+        useEffect(() => {
+            fetchScatter().then(data => {
+                setScatterData(data);
+            });
+        }, [activeResult, fetchScatter]);
 
-  useEffect(() => {
-    if (! R.isNil(selectedFeature)){
-      const prev = scatterData;
-      setScatterData( [] ) // loading 
-      fetchScatter().then((data) => {
-        const mapIndexed = R.addIndex(R.map)
-        const merged = R.ifElse(
-          R.has('error'),
-          () => {console.error(data['error']); return prev}, // show error message here
-          mapIndexed((val, index) => {
-            return R.mergeLeft(val, scatterData[index])
-          })
-        )(data)
-        setScatterData(merged)
-      })
-    }
-  }, [selectedFeature])
+        useEffect(() => {
+            if (selectedGroup) {
+                setScatterData([]); // set to loading
+                fetchScatter().then(data => {
+                    setScatterData(data);
+                });
+            }
+        }, [fetchScatter, selectedGroup]);
+
+        useEffect(() => {
+            if (!R.isNil(selectedFeature)) {
+                const prev = scatterData;
+                setScatterData([]); // loading
+                fetchScatter().then(data => {
+                    const mapIndexed = R.addIndex(R.map);
+                    const merged = R.ifElse(
+                        R.has("error"),
+                        () => {
+                            console.error(data["error"]);
+                            return prev;
+                        }, // show error message here
+                        mapIndexed((val, index) => {
+                            return R.mergeLeft(val, scatterData[index]);
+                        })
+                    )(data);
+                    setScatterData(merged);
+                });
+            }
+        }, [fetchScatter, scatterData, selectedFeature]);
```

Comment:
```suggestion
        }, [selectedFeature]);
```
For some reason the linter added more arguments to this hook. Since scatterData is updated by this useEffect function, it can't be passed as an argument or the function will run indefinitely. Also, fetchScatter is a thunk so doesn't need to be passed as it doesn't change